### PR TITLE
SeedDB script is working now

### PR DESCRIPTION
**Pull request by phuchle**
**What it does:**
Connects to mongoDB and clears out any existing documents, then throws down new data.  Currently only 3 mock users are implemented.

**Usage:** 

1.  Turn on MongoDB on your local. 
Ubuntu: `sudo service mongod start`
Mac: `mongod`

2.  `cd src/server`
`node seedDB.js`
See success messages logged to console and script exiting.

3. Start mongo shell with `mongo`

4. In mongo shell: 
`show dbs` => expect `aaf` to be present
`use aaf` to switch to the db
`show collections` => expect users, donors, families, names, organizers to be present.
`db.names.find()` to return names.
`db.names.find().length()` to be 3
`db.users.find().length()` to be 3
`db.donors.find().length()` to be 1
etc for other models.

5. Poke at it as much as you want.

**Methods:**
There is a liberal use of async/await in order to get this script to work in a 'synchronous' manner.  The `await` keyword can be used on anything that returns a promise and must be used inside an `async` function.  `await` allows you to pause the execution of your code until the promise comes back with data or errors out, essentially telling the program to wait for the promise to be done.  This allows the script to run asynchronously but look synchronous.

The mongoose methods I'm using in the script are all asynchronous so that's why there are so many cases of async/await.

I had to set the ecmaVersion to 2017 in order to use async/await.  Make sure you're on Node 8+ for it to work properly.

**Slops:**
I also just realized I left a comment in on seedDB.js line 216 that is no longer relevant.
Prettier + autosave wrecked me and I have some noise in the model files and eslintrc.json.  We should probably standardize prettier settings as well.
Wishlist not created yet, as I think this is best done when simulating when families adding items to the db.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,14 +1,12 @@
...
+    "ecmaVersion": 2017,
     "sourceType": "module",
     "ecmaFeatures": {
       "jsx": true
     }
   },
-  "plugins": [
-    "react"
-  ],
+  "plugins": ["react"],
```
Comment (psyduc) [inline .eslintrc.json:9]: just curious but why did you change this to inline?

Comment (jwu910) [reply .eslintrc.json:9]: Only one item, can fit in one line

Comment (phuchle) [reply .eslintrc.json:9]: It's a prettier preset.  I think 80 characters is the limit for making line breaks.  All the formatting changes were auto-saved by prettier so there is a lot of noise :(

Comment (psyduc) [reply .eslintrc.json:9]: gotcha, thanks

Comment (jwu910) [reply .eslintrc.json:9]: Does this prettier preset remove use of arrow functions?

Comment (phuchle) [reply .eslintrc.json:9]: No it doesn't, but it defaults to a style of `param => { ... }` instead of `(param) => { ... }`.  The methods on the models use a traditional function because I believe those methods will need `this` to point to the model in order to do stuff.

Comment (phuchle) [reply .eslintrc.json:9]: For multiple params, it will always be in `(param1, param2) => {}` because javascript requires that.

Comment (jwu910) [reply .eslintrc.json:9]: I see, i saw some arrow functions on the other files. i saw one change from () => {} to function() {}.

Comment (phuchle) [reply .eslintrc.json:9]: In most cases, we can use arrow functions but there are a few edge cases where we need a `function` declaration in order to access `this` correctly or accessing `arguments` within the function.

File: server/seedDB.js [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+  userOptions
+) => {
+  const nameModel = new Name(name);
+
+  await nameModel.save((err, name) => {
+    if (err) {
+      console.error(`Error creating Name.  Error: ${err}`);
+      return;
+    }
+    console.log('Successfully created name');
```
Comment (jwu910) [inline server/seedDB.js]: If you're passing `err` and `name` into this await, should you be still using name?

Should this line be 
```
console.log(`Successfully created ${name}`);
``` 
?

File: .eslintrc.json (hunk 1 above)
Comment (jwu910) [reply .eslintrc.json:9]: Still figuring those edge cases out. I remember some guys running into issues at work with 'this' binding in es5.

File: server/seedDB.js (hunk 2 above)
Comment (psyduc) [reply server/seedDB.js]: if you're using ${name} , then aren't you using name?

Comment (phuchle) [reply server/seedDB.js]: I think it should be something like ```console.log(`Successfully created name: ${name}`);```  Similarly with other `.save()` method calls (will update).

Comment (jwu910) [reply server/seedDB.js]: Sounds good. I can agree with
```
console.log(`Successfully created name: ${name}`);
```

Would make more sense than `Successfully created Josh.`

**Review by jwu910 — CHANGES_REQUESTED** [review]
I think just changing the log syntax, everything else seems to make sense thus far.

**Comment by phuchle** [comment]
Made the requested changes!

File: server/models/user.js [hunk 3]
```diff
@@ -8,33 +8,33 @@ const userSchema = new Schema({
...
     type: Schema.Types.ObjectId,
-    ref: 'Organizer'
-  }
+    ref: 'Organizer',
+  },
 });
 
 // Hash password before saving
-userSchema.pre('save', next => {
+userSchema.pre('save', function(next) {
```
Comment (rainrivas) [inline server/models/user.js:37]: we losing the arrow functions? shouldn't this be `(next) =>`

Comment (rainrivas) [reply server/models/user.js:37]: As per the official mongoose docs:
> Do not declare methods using ES6 arrow functions (=>). Arrow functions explicitly prevent binding this, so your method will not have access to the document and the above examples will not work.
[mongoose docs: methods](http://mongoosejs.com/docs/guide.html#methods)

**Review by rainrivas — APPROVED** [review]

**Review by jwu910 — APPROVED** [review]

**Review by psyduc — APPROVED** [review]
